quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PersistedConfigSource.java

```diff
@@ -43,6 +45,14 @@ public final class PersistedConfigSource extends PropertiesConfigSource {
     public static final String PERSISTED_PROPERTIES = "META-INF/keycloak-persisted.properties";
     private static final PersistedConfigSource INSTANCE = new PersistedConfigSource();
 
+    /**
+     * MicroProfile Config does not allow removing a config source when resolving properties. In order to be able
+     * to resolve the current (not the persisted value) value for a property, even if not explicitly set at runtime, we need
+     * to ignore this config source. Otherwise, default values are not resolved at runtime because the property will be
+     * resolved from this config source, if persisted.
+     */
+    private static final ThreadLocal<Boolean> ENABLED = new ThreadLocal<>();
```

Comment:
A thread local is needed because build steps run using multiple threads. Although the code relying on this thread local is never called during re-augmentation (when build steps are run) but only during the start (single-threaded), it is safer and should avoid concurrency issues when using this method at the wrong place.